platforms/software/testing-base-infrastructure/src/main/java/org/gradle/api/internal/tasks/testing/DefaultTestDescriptor.java

```diff
@@ -35,16 +37,19 @@ public DefaultTestDescriptor(Object id, @Nullable String className, String name)
 
     @UsedByScanPlugin("test-distribution")
     public DefaultTestDescriptor(Object id, @Nullable String className, String name, @Nullable String classDisplayName, String displayName) {
-        super(id, name);
+        this(id, className, name, classDisplayName, displayName, TestSources.missing());
```

Comment:
❗ IDK enough about the scan plugin to know if it supplying `missing` is going to be an issue.